document/chapters/preliminaries.tex

```diff
@@ -118,8 +118,23 @@ \subsection{Lifetimes}
 In principle, every reference needs a lifetime annotation. However, this is avoided by a process known as lifetime elision, where lifetimes are inferred automatically by the compiler. Even then, in certain cases the programmer might need to add such annotations. However, this will be discussed in the generics section.
 
 \section{Algebraic data types}
+Rust has both sum and product algebraic data types in the form of structures and enumerations respectively.
```

Comment:
```suggestion
Rust has both sum and product algebraic data types~\cite{REF} in the form of structures and enumerations respectively.
```

Reply:
We need a reference about the notion of algebraic data types? or about the fact that Rust has algebraic data types?